### ISSUE TITLE
Field error not sync between onValidate() method and UI

### ISSUE THREAD

**Issue by chwizdo** [edmundhung/conform#344]
### Describe the bug and the expected behavior

Field validation error not sync between the `onValidate()` method in `useForm()` parameter and the UI. In the `onValidate()` method body, the `date` field after parsed has error, however in the UI, `date.errors` is `undefined`. The expected behaviour is `date.errors` should contains error (specifically an array of an item: "Required") if the form is submitted without picking a date.

### Conform version

0.9.1

### Steps to Reproduce the Bug or Issue

1. Create a new Remix project
2. Install zod and conform dependencies
3. Create a new route/page, and add the codes below the instructions
4. Submit the form without picking a date

```typescript
import { conform, useForm, useInputEvent } from "@conform-to/react";
import { getFieldsetConstraint, parse } from "@conform-to/zod";
import { z } from "zod";
import { Form } from "@remix-run/react";
import { useId, useRef } from "react";

const IncomeGenerateSchema = z.object({ date: z.date() });

const ExamplePage = () => {
  const formId = useId();

  const [form, { date }] = useForm({
    id: formId,
    constraint: getFieldsetConstraint(IncomeGenerateSchema),
    onValidate: ({ formData }) => {
      const parsed = parse(formData, { schema: IncomeGenerateSchema });
      // Console log result: ["Required"]
      console.log(parsed.error["date"]);
      return parsed;
    },
  });

  const shadowInputRef = useRef<HTMLInputElement>(null);
  const control = useInputEvent({ ref: shadowInputRef });

  // Console log result: undefined
  console.log(date.errors);

  return (
    <Form method="POST" className="p-4" {...form.props}>
      <input onChange={control.change} type="date" />
      <input
        ref={shadowInputRef}
        {...conform.input(date, { type: "hidden" })}
      />
      {date.errors && (
        <ul>
          {date.errors.map((error, index) => (
            <li key={index}>
              <p>{error}</p>
            </li>
          ))}
        </ul>
      )}
      <button>Submit</button>
    </Form>
  );
};

export default ExamplePage;
```

### What browsers are you seeing the problem on?

Others

**Comment by edmundhung** [comment]
Hi @chwizdo, I believe the issue you had is caused by the hidden type input. If you use the hidden attribute instead (i.e. `<input hidden />`), it should work. This is a limitation caused by the use of the `setCustomValidity` API at the moment and it will be addressed when we release the changes described on this [RFC](https://github.com/edmundhung/conform/discussions/331).

**Comment by chwizdo** [comment]
@edmundhung Yes, by using the hidden attribute instead, it'll work. I understand now, I'll temporary use this workaround for the time being. However, do u know the consequences of this workaround? What's the differences of using the hidden property and the hidden type?

**Comment by edmundhung** [comment]
This is just how the Constraint Validation APIs works unfortunately. I have read the HTML spec about this and my understanding is that `<input type="hidden" />` was mainly used as a way for the server to provide some additional context to the client. The browser believes the value would never change and so there is no need to do any client validation. It's just that the community find it useful to hide an input when implementing a custom input so we were all using it. 

However, if we use `<input hidden />`, we are actually creating an input with type `text`. As the constraint validation API do not have specific instructions about the `hidden` attribute, it just being treated as any text input.

**Comment by edmundhung** [comment]
This should be fixed in v1.0.0.
